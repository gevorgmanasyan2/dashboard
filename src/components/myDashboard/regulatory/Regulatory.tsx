/* eslint-disable @typescript-eslint/camelcase */
/* eslint-disable react/display-name */
import React, { FC, useEffect } from 'react'
import { connect, useDispatch } from 'react-redux'
import { MessageFilled, WarningFilled } from '@ant-design/icons'
import IsLoading from '../../../spinner/IsLoading'

import { Table, Tag } from 'antd'
import { fetchWaiverData } from '../../../redux/waiver/actions'
import { Link } from 'react-router-dom'
import './regulatory.scss'

interface RegulatoryProps {
  isLoading?: boolean
  inputValue?: string
  waivers?: waiverType[]
}

type waiverType = {
  id: number
  name: string
  percent_completion: number
  open_questionnaires: number
  waiver_type_name: string
  waiver_status_name: string
  assigned_to_user_name: string
}

const Regulatory: FC<RegulatoryProps> = ({ waivers, isLoading, inputValue }) => {
  const tableData = waivers?.map((row) => ({
    key: row.id,
    name: row.name,
    type: row.waiver_type_name,
    tag: row.waiver_status_name,
    percent: `${row.percent_completion}${'%'}`,
    assigned: row.assigned_to_user_name,
    flag: row.open_questionnaires,
  }))
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(fetchWaiverData())
  }, [dispatch])

  const columns = [
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
      render: (row: any, record: any) => {
        return <Link to={`/regulatory-complete-document/${record.key}`}>{row}</Link>
      },
    },
    {
      title: 'Type',
      dataIndex: 'type',
      key: 'type',
    },
    {
      title: 'Status',
      dataIndex: 'tag',
      key: 'tag',
      render: (tag: any) => (
        <Tag color={tag === 'open' ? 'green' : 'geekblue'} key={tag}>
          {tag}
        </Tag>
      ),
    },
    {
      title: 'Assigned Agent',
      dataIndex: 'assigned',
      key: 'assigned',
    },
    {
      title: 'Percent',
      dataIndex: 'percent',
      key: 'percent',
    },
    {
      title: 'Action flag',
      dataIndex: 'flag',
      key: 'flag',
      render: (text: any) => {
        return text > 0 ? <WarningFilled className="warning-icon" /> : <MessageFilled />
      },
    },
  ]

  const filtered = tableData?.filter((waiver: any) =>
    waiver.name.toLowerCase().trim().includes(inputValue?.toLowerCase().trim())
  )
  return (
    <div className="regulatory__items">
      {isLoading ? (
        <IsLoading />
      ) : (
        <Table
          dataSource={inputValue ? filtered : tableData}
          columns={columns}
          bordered={true}
          scroll={{ x: '100%' }}
          
        />
      )}
    </div>
  )
}

const mapStateToProps = (state: any) => {
  return {
    isLoading: state.waiver.isLoading,
    waivers: state.waiver.waivers,
  }
}

export default connect(mapStateToProps)(Regulatory)
