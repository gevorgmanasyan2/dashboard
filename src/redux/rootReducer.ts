import { combineReducers } from 'redux'
import * as auth from '../redux/auth/authReducer'
import * as data from './count-data/fetchDataReducer'
import * as waivers from './waiver/fetchWaiverDataReducer'
import * as location from './location/locationReducer'
import * as messages from './messages/messagesReducer'
import * as user from '../redux/my-account/userReducer'
import { authReducer } from './auth/authReducer'
import { fetchDataReducer } from './count-data/fetchDataReducer'
import { fetchWaiverDataReducer } from './waiver/fetchWaiverDataReducer'
import { locationReducer } from './location/locationReducer'
import { messagesReducer } from './messages/messagesReducer'
import { userReducer } from './my-account/userReducer'
export interface CombinedState {
  auth: auth.State
  data: data.State
  waivers: waivers.State
  location: location.State
  messages: messages.State
  user: user.State
}
const rootReducer = combineReducers({
  auth: authReducer,
  data: fetchDataReducer,
  waiver: fetchWaiverDataReducer,
  location: locationReducer,
  messages: messagesReducer,
  user: userReducer,
})

export type RootState = ReturnType<typeof rootReducer>
export default rootReducer
