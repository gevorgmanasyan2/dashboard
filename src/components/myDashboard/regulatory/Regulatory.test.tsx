/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import Regulatory from './Regulatory'
import { Provider } from 'react-redux'
import { store } from '../../../redux/store'

it('should render Regulatory component', () => {
  const mockItem = {
    waivers: [],
    isLoading: true,
  }
  expect(
    shallow(
      <Provider store={store}>
        <Regulatory {...mockItem} />
      </Provider>
    )
  ).toMatchSnapshot()
})
