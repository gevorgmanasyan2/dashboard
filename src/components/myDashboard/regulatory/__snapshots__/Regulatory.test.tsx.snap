// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Regulatory component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Regulatory)
    isLoading={true}
    waivers={Array []}
  />
</ContextProvider>
`;
